### PR TITLE
[bug_fix] Fix importing for mso_schema_template_anp_epg_contract to select the contract with the correct type (DCNE-205)

### LINKED ISSUE #298: mso_schema_template_anp_epg_contract import not supporting same contract provided and consumed (DCNE-205)
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

import of mso_schema_template_anp_epg_contract doesnt take the relationship type as an argument so in case same contract is consumed and provided on an epg, we are unable to import consumer side

### New or Affected Resource(s) + MSO Class(es):

* mso_schema_template_anp_epg_contract

### MSO version

* V 4.2.3

### APIC version and APIC Platform for Site level Resources

* V 5.2/6.0

### Potential Terraform Configuration

```hcl
 terraform import mso_schema_template_anp_epg_contract.epg1_contract1_consumed <site id>/template/<template>/anp/<AP>/epg/<EPG>/contract/CONTRACT
```

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #298

### CODE REVIEW

**Review by anvitha-jain — COMMENTED** [review]
Is this a modification in certain version? If I remember, at one point the contract could have both consumer and provider relationship. Did this change in recent version?

**Comment by akinross** [comment]
> Is this a modification in certain version? If I remember, at one point the contract could have both consumer and provider relationship. Did this change in recent version?

I see when looking at git blame that the relationship_type is already part of the original code from 4 years ago and is also part of the read function. I assumed ( perhaps wrongly ) that since it is already done also in the read function this is missing in import only by mistake. If it was introduced at a later stage in NDO/MSO then I think it is already broken for the read function for a while, and thus the resource should not be working properly in older versions.

I can look up when this is introduced, will look through release notes and get back to you.

**Comment by shrsr** [comment]
@anvitha-jain Are you referring to the the resource in Terraform ACI that you worked on?
https://github.com/CiscoDevNet/terraform-provider-aci/commit/1466a6d9a9487ad767038bb5d494e77c85f71858

**Review by shrsr — DISMISSED** [review]
LGTM

**Review by gmicol — DISMISSED** [review]

**Review by sajagana — APPROVED** [review]
LGTM!

**Comment by lhercot** [comment]
I think we are mixing it up with the template contract resource filterRelationships, filterRelationshipsConsumerToProvider and filterRelationshipsProviderToConsumer attributes which are separate lists.

The only issue that could exist here is that there are other values for relationship_type in the API documentation and we are not listing them all.

File: website/docs/r/schema_template_anp_epg_contract.html.markdown [hunk 1]
```diff
@@ -45,5 +45,5 @@ No attributes are exported.
 An existing MSO Schema Template Application Network Profile Endpoint Group Contract can be [imported][docs-import] into this resource via its Id/path, via the following command: [docs-import]: <https://www.terraform.io/docs/import/index.html>
 
 ```bash
-terraform import mso_schema_template_anp_epg_contract.contract1 {schema_id}/template/{template_name}/anp/{anp_name}/epg/{epg_name}/contract/{contract_name}
+terraform import mso_schema_template_anp_epg_contract.contract1 {schema_id}/template/{template_name}/anp/{anp_name}/epg/{epg_name}/contract/{contract_name}/relationship_type/{consumer|provider}
```
Comment (lhercot) [inline website/docs/r/schema_template_anp_epg_contract.html.markdown:48]:
```suggestion
terraform import mso_schema_template_anp_epg_contract.contract1 {schema_id}/template/{template_name}/anp/{anp_name}/epg/{epg_name}/contract/{contract_name}/relationshipType/{consumer|provider}
```

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by samiib — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**